pareto/models_extra/Integrate_desal/models/qcp_desal.py

```diff
@@ -0,0 +1,633 @@
+
+    # Minimum treatment concentration required
+    model.p_Cmin = pyo.Param(
+        model.s_NTCW,
+        model.s_Q,
+        initialize=data["p_Cmin"],
+        mutable=True,
+        doc="Minimum concentration required at the concentrated water node",
+        within=pyo.Reals,
+    )
+    model.p_Fmin = pyo.Param(
+        model.s_NTIN,
+        initialize=data["p_Fmin"],
+        mutable=True,
+        doc="Minimum Flow required at each treatment site",
+        within=pyo.Reals,
+    )
+
+    # Cost Parameters
+    model.p_betaArc = pyo.Param(
+        model.s_A,
+        initialize=data["p_betaArc"],
+        doc="Operational costs of arcs",
+        within=pyo.Reals,
+    )
+    model.p_betaD = pyo.Param(
+        model.s_ND,
+        initialize=data["p_betaD"],
+        mutable=True,
+        doc="Operational cost of disposal site",
+        within=pyo.Reals,
+    )
+    model.p_betaW = pyo.Param(
+        model.s_NW,
+        initialize=data["p_betaW"],
+        doc="Costs of sourcing freshwater",
+        within=pyo.Reals,
+    )
+    model.p_betaT = pyo.Param(
+        model.s_NTIN,
+        initialize=data["p_betaT"],
+        mutable=True,
+        doc="Operating cost of treatment facility",
+        within=pyo.Reals,
+    )
+    model.p_betaS = pyo.Param(
+        model.s_NS,
+        initialize=data["p_betaS"],
+        mutable=True,
+        doc="Costs of storing produced water",
+        within=pyo.Reals,
+    )
+    model.p_betaR = pyo.Param(
+        model.s_NTTW,
+        initialize=data["p_betaR"],
+        doc="Cost of beneficial reuse operation",
+        within=pyo.Reals,
+    )
+
+    model.p_gammaS = pyo.Param(
+        model.s_NS,
+        initialize=data["p_gammaS"],
+        mutable=True,
+        doc="Earnings from retrieving water from storage tanks",
+        within=pyo.Reals,
+    )
+    model.p_gammaT = pyo.Param(
+        model.s_NTCW,
+        model.s_Q,
+        initialize=data["p_gammaT"],
+        doc="Earnings from each component",
+        within=pyo.Reals,
+    )
+    model.p_gammaR = pyo.Param(
+        model.s_NTTW,
+        initialize=data["p_gammaR"],
+        doc="Earnings from beneficial reuse",
+        within=pyo.Reals,
+    )
+
+    # Functions
+    model.p_nodeUp = pyo.Param(
+        model.s_A,
+        initialize=data["p_nodeUp"],
+        doc="Upstream node of the arc",
+        within=pyo.Any,
+    )
+    model.p_nodeDown = pyo.Param(
+        model.s_A,
+        initialize=data["p_nodeDown"],
+        doc="Downstream node of the arc",
+        within=pyo.Any,
+    )
+    model.p_tp = pyo.Param(
+        model.s_A, initialize=data["p_tp"], doc="Type of arc", within=pyo.Any
+    )
+    model.p_treatedIN = pyo.Param(
+        model.s_NTTW | model.s_NTCW,
+        initialize=data["p_treatedIN"],
+        doc="Inlet node for each treatment facility",
+        within=pyo.Any,
+    )
+
+    # --------------------VARIABLES-----------------------------
+    # TODO: Do the bounds properly
```

Comment:
Does this TODO need to be addressed? Please either address it or remove the comment.

Reply:
@Sakshi21299 can you please weigh in here? Alternatively, @JavalVyas2000 do you know the answer?

Reply:
I addressed this in 03d6074f05193d4e62ca4d95f9b254d817238d12 - resolving.